packages/front-end/components/Features/SDKConnections/SDKConnectionForm.tsx

```diff
@@ -0,0 +1,186 @@
+  initialValue = {},
+  edit,
+  close,
+  mutate,
+}: {
+  initialValue?: Partial<SDKConnectionInterface>;
+  edit: boolean;
+  close: () => void;
+  mutate: () => void;
+}) {
+  const environments = useEnvironments();
+  const { project, projects } = useDefinitions();
+  const { apiCall } = useAuth();
+  const router = useRouter();
+
+  const [upgradeModal, setUpgradeModal] = useState(false);
+
+  const form = useForm({
+    defaultValues: {
+      name: initialValue.name || "",
+      languages: initialValue.languages || [],
+      environment: initialValue.environment || environments[0]?.id || "",
+      project: "project" in initialValue ? initialValue.project : project || "",
+      encryptPayload: initialValue.encryptPayload || false,
+      proxyEnabled: initialValue.proxy?.enabled || false,
+      proxyHost: initialValue.proxy?.host || "",
+    },
+  });
+
+  if (upgradeModal) {
+    return (
+      <UpgradeModal
+        close={() => setUpgradeModal(false)}
+        reason="To enable SDK encryption,"
+        source="encrypt-features-endpoint"
+      />
+    );
+  }
+
+  const hasSDKsWithoutEncryptionSupport = form
+    .watch("languages")
+    .some((l) => !languageMapping[l].supportsEncryption);
+
+  return (
+    <Modal
+      header={edit ? "Edit SDK COnnection" : "New SDK Connection"}
+      size={"lg"}
+      submit={form.handleSubmit(async (value) => {
+        if (edit) {
+          const body: EditSDKConnectionParams = {
+            name: value.name,
+            languages: value.languages,
+            proxyEnabled: value.proxyEnabled,
+            proxyHost: value.proxyHost,
+          };
+          await apiCall(`/sdk-connections/${initialValue.id}`, {
+            method: "PUT",
+            body: JSON.stringify(body),
+          });
+          mutate();
+        } else {
+          // Make sure encryption is disabled if they selected at least 1 language that's not supported
+          // This is already be enforced in the UI, but there are some edge cases that might otherwise get through
+          // For example, toggling encryption ON and then selecting an unsupported language
+          if (
+            value.languages.some((l) => !languageMapping[l].supportsEncryption)
+          ) {
+            value.encryptPayload = false;
+          }
+
+          const body: Omit<CreateSDKConnectionParams, "organization"> = {
+            ...value,
+          };
+          const res = await apiCall<{ connection: SDKConnectionInterface }>(
+            `/sdk-connections`,
+            {
+              method: "POST",
+              body: JSON.stringify(body),
+            }
+          );
+
+          mutate();
+          await router.push(`/sdks/${res.connection.id}`);
+        }
+      })}
+      close={close}
+      open={true}
+      cta="Save"
+    >
+      <Field label="Name" {...form.register("name")} required />
+
+      <div className="form-group">
+        <label>Tech Stack</label>
```

Comment:
initial form has:
<label>Tell us a little about your tech stack</label>

Should we (a) use the same language for both, and (b) explain that this really only is used for documentation instructions (other than JS streaming)?